docs/semgrep-code/notifications.md

```diff
@@ -402,40 +403,146 @@ If you are using GitHub Actions to run Semgrep, no extra changes are needed to g
+#### Creating and adding a repository access token
+
+:::info Prerequisites
+- Configured pipeline in Bitbucket CMS.
+- Established connection between Bitbucket and Semgrep Cloud Platform. The **SEMGREP_APP_TOKEN** added as a **repository** variable in Bitbucket. See the [Creating a SEMGREP_APP_TOKEN](/semgrep-ci/running-semgrep-ci-with-semgrep-cloud-platform/#creating-a-semgrep_app_token) section and [Repository variables](https://support.atlassian.com/bitbucket-cloud/docs/variables-and-secrets/#Repository-variables) in Bitbucket documentation.
+:::
 
 :::note
-GitLab MR comments are only available to logged-in Semgrep users, as they require a Semgrep API token.
+This section helps you to configure PR comments if you do **not** have the Bitbucket Cloud Premium plan. You can create a separate repository access token for each repository where you want to use Semgrep. This configuration option is also useful if you have the Bitbucket Cloud Premium plan, but prefer to onboard repositories one by one instead of bulk onboarding.
+:::
+
+Fulfill these general steps to create a repository access token:
+
+1. Create a repository access token in Bitbucket with **Read**, and **Write** permissions for pull requests. Follow the instructions in [Create a repository Access Token](https://support.atlassian.com/bitbucket-cloud/docs/create-a-repository-access-token/) in Bitbucket documentation.
+1. Add the repository access token as a repository variable with the **Secured** option.
+
+To complete the configuration, follow the [Adding Semgrep to your Bitbucket CI pipeline for PR comments](/semgrep-code/notifications/#bitbucket-ci-pipelines-yaml-file-for-pr-comments) section.
+
+#### Bitbucket CI pipelines YAML file for PR comments
+
+:::info Prerequisite
+Finish the steps in either [Creating a workspace access token](/semgrep-code/notifications/#creating-and-adding-a-workspace-access-token) (if you have **Bitbucket Cloud Premium** plan), or [Creating a repository access token](/semgrep-code/notifications/#creating-and-adding-a-repository-access-token) before committing the file below.
+:::
+
+Commit the following code to your `bitbucket-pipelines.yml` file:
+
+```yaml
+image: atlassian/default-image:latest
+
+pipelines:
+  branches:
+    main:
+        - step:
+            name: 'Run Semgrep full scan with main branch'
+            image: returntocorp/semgrep
+            script:
+              - export SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN
+              - semgrep ci
+  pull-requests:
+    '**':
+      - step:
+          name: 'Run Semgrep diff scan with PR branch'
+          image: returntocorp/semgrep
+          script:
+            # Set SEMGREP Variables
+            - export SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN
+            - export SEMGREP_BASELINE_REF="origin/main"
+            - git fetch origin "+refs/heads/*:refs/remotes/origin/*"
+            - export BITBUCKET_TOKEN=$PAT
+            - semgrep ci
+```
```

Comment:
Instead of putting this here, we might just want to say that we have to export the PAT as BITBUCKET_TOKEN. I think the key here is just to export the token as BITBUCKET_TOKEN so Semgrep can recognize it and use it to authenticate with the Bitbucket API. 

Instead of putting this config file here since it's specific only to Bitbucket Pipelines, we might want to move it here instead and replace the current config file recommended: https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-bitbucket-pipelines-configuration-snippet

The current config file only recommends running full scans on all branches so we might want to replace that with this config while still keeping the other information about potential CI variables that can be set (e.g. SEMGREP_JOB_URL, SEMGREP_COMMIT etc.)

Reply:
I volunteer to move it in a follow up PR (since I will have to look at the CI docs anyway) :D We can merge it for now and I'll write the update immediately 
 after.  cc @pangngan

Reply:
Awesome, sounds good to me! Thank you all!!

Reply:
Feel free @s-santillan to push to this branch also - whatever is easier for you. :-)

Reply:
Generally pushing into other contributor branches is welcomed practice I think. ;-) @s-santillan

Reply:
@s-santillan @pangngan The only downside I see with linking to the code snippet in https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-bitbucket-pipelines-configuration-snippet is that we need the PR part uncommented here (this one right? `- git fetch origin "+refs/heads/*:refs/remotes/origin/*"`) but in the Sample configs, we should make it optional and commented out (`# - git fetch origin "+refs/heads/*:refs/remotes/origin/*"), so it creates a bit more confusion here - open this link, copy the code snippet and uncomment this line... it's not easy. For the rest (token), feel free to push to this branch, I will don't do any other edits here except for accepting your suggestions. 🙏

Reply:
@adamkvitek The pattern “open this link, copy the code snippet and uncomment this line” is well understood.

Reply:
@pabloest That's why it would be great to have conditional text in a module (such as https://docs.asciidoctor.org/asciidoc/latest/directives/conditionals/) and import a module here with a different condition than in another text. Creating documentation which is a collection of links is not great.

Reply:
Let's figure out how to solve these issues later. Resolving and merging for now.